lib/msal-common/src/auth/IdToken.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ClientAuthError } from "../error/ClientAuthError";
+import { StringDict } from "../utils/MsalTypes";
+import { StringUtils } from "../utils/StringUtils";
+import { ICrypto } from "../utils/crypto/ICrypto";
+
+/**
+ * @hidden
+ */
+export class IdToken {
+
+    issuer: string;
+    objectId: string;
+    subject: string;
+    tenantId: string;
+    version: string;
+    preferredName: string;
+    name: string;
+    homeObjectId: string;
+    nonce: string;
+    expiration: string;
+    rawIdToken: string;
+    claims: StringDict;
+    sid: string;
+    cloudInstance: string;
+    /* tslint:disable:no-string-literal */
+    constructor(rawIdToken: string, crypto: ICrypto) {
+        if (StringUtils.isEmpty(rawIdToken)) {
+            throw ClientAuthError.createIdTokenNullOrEmptyError(rawIdToken);
+        }
+
+        this.rawIdToken = rawIdToken;
+        this.claims = IdToken.extractIdToken(rawIdToken, crypto);
+        if (this.claims.hasOwnProperty("iss")) {
+            this.issuer = this.claims["iss"];
+        }
+
+        if (this.claims.hasOwnProperty("oid")) {
+            this.objectId = this.claims["oid"];
+        }
+
+        if (this.claims.hasOwnProperty("sub")) {
+            this.subject = this.claims["sub"];
+        }
+
+        if (this.claims.hasOwnProperty("tid")) {
+            this.tenantId = this.claims["tid"];
+        }
+
+        if (this.claims.hasOwnProperty("ver")) {
+            this.version = this.claims["ver"];
+        }
+
+        if (this.claims.hasOwnProperty("preferred_username")) {
+            this.preferredName = this.claims["preferred_username"];
+        }
+
+        if (this.claims.hasOwnProperty("name")) {
+            this.name = this.claims["name"];
+        }
+
+        if (this.claims.hasOwnProperty("nonce")) {
+            this.nonce = this.claims["nonce"];
+        }
+
+        if (this.claims.hasOwnProperty("exp")) {
+            this.expiration = this.claims["exp"];
+        }
+
+        if (this.claims.hasOwnProperty("home_oid")) {
+            this.homeObjectId = this.claims["home_oid"];
+        }
+
+        if (this.claims.hasOwnProperty("sid")) {
+            this.sid = this.claims["sid"];
+        }
+
+        if (this.claims.hasOwnProperty("cloud_instance_host_name")) {
+            this.cloudInstance = this.claims["cloud_instance_host_name"];
+        }
+        /* tslint:enable:no-string-literal */
+    }
+
+    /**
+     * Extract IdToken by decoding the RAWIdToken
+     *
+     * @param encodedIdToken
+     */
+    static extractIdToken(encodedIdToken: string, crypto: ICrypto): any {
```

Comment:
`ClientInfo` on the return type correct?

Reply:
no it would be an object with id token parameters. We can fully define Id token if that would make it easier to define this, but for now we just parse the id token object as a JSON.

Reply:
See above comment.